_posts/2020-06-23-user-feedback-websockets.md

```diff
@@ -0,0 +1,289 @@
+---
+layout: post
+authors: [kevin_van_houtte]
+title: 'Enabling User Feedback with WebSockets on RabbitMQ and Spring Cloud'
+image: /img/sockets/feedback.jpeg
+tags: [spring, queue, integration, websockets,  event-driven, cloud]
+category: Event-Driven
+comments: true
+---
+
+### Reading time: 7 minutes and 31 seconds
+
+When working with event-drive applications, you tend to see this on the screen:
+
+> We are processing your request, we will notify you when it succeeds
+
+Let's chop the sentence down into processing the request...
+This means that the server is processing your request, but the client is not sure if it was entirely successful because it got transformed into an event and published on a queue never to be seen again (fire and forget).  
+The client wonders what has happened and needs a way to give his users the state of the request.  
+The next statement tells the user it will notify him/her when the request was successful.  
+This part can get complex because you want to give a rapid response as soon as possible.  
+
+>The first thing that comes to mind is, can our client not pol the state of the data from the table un till it's ready?   
+
+When our client receives a high amount of load, and it's polling the database for the state of the data, it can put it under unnecessary stress.  
+Since polling is a periodically check, it is not real-time, and we want to bring feedback to our users as soon as possible.  
+To let your client behave in real-time, we need push events.  
+Push events can be enabled by the concept of WebSockets, this bilateral communication connects the server and the client in an open connection with each other.  
+This tech post will explain how I enabled push events with RabbitMQ, MQTT, and Spring Cloud Stream.  
```

Comment:
We do not talk in first person, always "we" -> how we enabled
or 
"This article will explain how to push events"